hub/api/dataset.py

```diff
@@ -255,24 +261,46 @@ def mode(self):
+        drop_last: Optional[bool] = False,
+        collate_fn: Optional[Callable] = None,
+        pin_memory: Optional[bool] = False,
     ):
-        """Converts the dataset into a pytorch compatible format.
+        """Converts the dataset into a pytorch Dataloader.
 
         Note:
             Pytorch does not support uint16, uint32, uint64 dtypes. These are implicitly type casted to int32, int64 and int64 respectively.
-            This spins up it's own workers to fetch data, when using with torch.utils.data.DataLoader, set num_workers = 0 to avoid issues.
+            This spins up it's own workers to fetch data.
 
         Args:
-            transform (Callable, optional) : Transformation function to be applied to each sample
-            workers (int): The number of workers to use for fetching data in parallel.
-            tensors (List, optional): Optionally provide a list of tensor names in the ordering that your training script expects. For example, if you have a dataset that has "image" and "label" tensors, if `tensors=["image", "label"]`, your training script should expect each batch will be provided as a tuple of (image, label).
+            transform (Callable, optional) : Transformation function to be applied to each sample.
+            num_workers (int): The number of workers to use for fetching data in parallel.
+            tensors (List, optional): Optionally provide a list of tensor names in the ordering that your training script expects.
+                For example, if the dataset that has "image" and "label" tensors and `tensors=["image", "label"]`, your training script should expect each batch will be provided as a tuple of (image, label).
+            batch_size (int, optional): Number of samples per batch to load. Default value is 1.
+            drop_last (bool, optional): Set to True to drop the last incomplete batch, if the dataset size is not divisible by the batch size.
+                If False and the size of dataset is not divisible by the batch size, then the last batch will be smaller. Default value is False.
+                Read torch.utils.data.DataLoader docs for more details.
+            collate_fn (Callable, optional): merges a list of samples to form a mini-batch of Tensor(s). Used when using batched loading from a map-style dataset.
+                Read torch.utils.data.DataLoader docs for more details.
+            pin_memory (bool, optional): If True, the data loader will copy Tensors into CUDA pinned memory before returning them. Default value is False.
+                Read torch.utils.data.DataLoader docs for more details.
 
         Returns:
-            A dataset object that can be passed to torch.utils.data.DataLoader
+            A torch.utils.data.DataLoader object.
         """
-        return dataset_to_pytorch(self, transform, workers=workers, tensors=tensors)
+        return dataset_to_pytorch(
+            self,
+            transform,
+            num_workers=num_workers,
```

Comment:
I don't like the idea of maintaining compatibility with all the params the dataloader might want to take. Maybe `**kwargs` can help here?

Reply:
we had this discussion with a few different people, @AbhinavTuli went through the args and out of the 13 we only can support like 4